### PR TITLE
docs: React core documentation

### PR DESCRIPTION
**Pull request by mGGk-fr**
# What

This MR add documentation for the `@front-commerce/core/react` public API

# Demo
https://deploy-preview-778--heuristic-almeida-1a1f35.netlify.app/docs/remixed/api-reference/front-commerce-core/react

### CODE REVIEW

File: docs/04-api-reference/front-commerce-core/react.mdx [hunk 1]
```diff
@@ -0,0 +1,197 @@
+---
+title: "React"
+sidebar_position: 3
+description: "This documentation will cover all React dedicated APIs."
```
Comment (real34) [inline docs/04-api-reference/front-commerce-core/react.mdx:4]:
```suggestion
description: "This documentation will cover all React dedicated APIs provided by the Front-Commerce core."
```

File: docs/04-api-reference/front-commerce-core/react.mdx [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+---
+
+<p>{frontMatter.description}</p>
+
+To help you creating the best front-end with React, we've created the some React
+piece of code that you can reuse in your components.
```
Comment (real34) [inline docs/04-api-reference/front-commerce-core/react.mdx:10]:
```suggestion
To help you creating the best front-end with React, we've created convenient React
hooks and components that you can reuse in your application.
```

File: docs/04-api-reference/front-commerce-core/react.mdx [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+    <div>
+      {isMounted ? "Component is mounted" : "Component isn't mounted yet"}
+    </div>
+  );
+};
+```
+
+## `useItemsValidityCheck`
+
+Return an object to tell the status of validty of an array
```
Comment (real34) [inline docs/04-api-reference/front-commerce-core/react.mdx:69]:
```suggestion
Return an object to tell the status of validty of an array (mostly used in the context of cart items).
```

File: docs/04-api-reference/front-commerce-core/react.mdx [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+  return (
+    <div>
+      Validity status : Are all items valid ? {arrayValidation.allValid}
+      Is array have at lease one valid item ? {arrayValidation.hasValid}
+      Are all item passed validity resolution:
+      {arrayValidation.allValidityResolved}
+    </div>
+  );
+};
+```
```
Comment (real34) [inline docs/04-api-reference/front-commerce-core/react.mdx:120]: **issue:**  I don't think it makes sense to invent an irrelevant example. This API is here for legacy reasons, and used only for adding items to cart/wishlist. I think we'd better get rid of it and I'd rather keep the doc succint.

**suggestion:** remove example, and add maybe a sentence saying that it's an API we don't recommend to use unless there is a good use case for it (or something like that :))

File: docs/04-api-reference/front-commerce-core/react.mdx [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+
+## `useCallbackModal`
+
+Manage actions with modal and modal state
```
Comment (real34) [inline docs/04-api-reference/front-commerce-core/react.mdx:124]:
```suggestion
Manage interactions with a modal and its state.
```

**Review by real34 — APPROVED** [review]

File: docs/04-api-reference/front-commerce-core/react.mdx (hunk 5 above)
Comment (mGGk-fr) [reply docs/04-api-reference/front-commerce-core/react.mdx]: fixed with your suggestion

File: docs/04-api-reference/front-commerce-core/react.mdx (hunk 4 above)
Comment (mGGk-fr) [reply docs/04-api-reference/front-commerce-core/react.mdx]: fixed

File: docs/04-api-reference/front-commerce-core/react.mdx (hunk 3 above)
Comment (mGGk-fr) [reply docs/04-api-reference/front-commerce-core/react.mdx]: fixed with your suggestion

File: docs/04-api-reference/front-commerce-core/react.mdx (hunk 2 above)
Comment (mGGk-fr) [reply docs/04-api-reference/front-commerce-core/react.mdx]: fixed with your suggestion

File: docs/04-api-reference/front-commerce-core/react.mdx (hunk 1 above)
Comment (mGGk-fr) [reply docs/04-api-reference/front-commerce-core/react.mdx]: fixed with your suggestion

**Pull request merged**